parser/core/models/tokens.go

```diff
@@ -0,0 +1,38 @@
+package models
+
+// TokenType represents the type of a token for syntax highlighting
+type TokenType string
+
+const (
+	// Literals
+	TokenString  TokenType = "string"  // String literals
+	TokenInteger TokenType = "integer" // Integer literals
+	TokenFloat   TokenType = "float"   // Float literals
+	TokenBoolean TokenType = "boolean" // Boolean literals
+
+	// Identifiers
+	TokenIdentifier TokenType = "identifier" // Column names
+	TokenFunction   TokenType = "function"   // Function names
+
+	// Operators
+	TokenOperator TokenType = "operator" // Arithmetic/comparison/logical operators
+
+	// Delimiters
+	TokenDelimiter TokenType = "delimiter" // Parentheses, commas, etc.
```

Comment:
Use other token type for parentheses. I want to highlight another paren when either of it hovered.